src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -825,6 +828,189 @@ void RigBundleAdjuster::ParameterizeCameraRigs(Reconstruction* reconstruction) {
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// PositionPriorBundleAdjuster
+////////////////////////////////////////////////////////////////////////////////
+
+PositionPriorBundleAdjuster::PositionPriorBundleAdjuster(
+    const BundleAdjustmentOptions& options,
+    const BundleAdjustmentConfig& config)
+    : BundleAdjuster(options, config) {
+  prior_options_.prior_position_covariance =
+      options_.prior_position_std.cwiseAbs2().asDiagonal();
+}
+
+bool PositionPriorBundleAdjuster::Solve(Reconstruction* reconstruction) {
+  loss_function_ =
+      std::unique_ptr<ceres::LossFunction>(options_.CreateLossFunction());
+
+  if (options_.use_robust_loss_on_prior_position) {
+    prior_loss_function_ =
+        std::make_unique<ceres::CauchyLoss>(options_.prior_position_loss_scale);
+  }
+
+  // Compute initial squared error between position priors & current images
```

Comment:
Instead of aligning again before each BA, wouldn't it be easier to somewhere store the `Sim3d` transformation between normalized and metric reconstructions? @ahojnnes @B1ueber2y as discussed previously, this could be stored in the `Reconstruction` object. Something like a Gage object:
```C++
struct Gage {
  std::optional<double> scale;  // metric scale of the reconstruction
  std::optional<Eigen::Vector3d> gravity;  // [0,0,-g] in the world coordinate frame
  std::optional<double> heading; // w.r.t East or North axis
  std::optional<Eigen::Vector3d> origin; // in ECEF or WGS84
}
```
This will be useful for IMU optimization too and would simplify managing the gage freedom of the BA (e.g. whether to fix the first pose or not).